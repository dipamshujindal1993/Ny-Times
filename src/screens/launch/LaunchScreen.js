import React from 'react'
import {View,TextInput,Text} from 'react-native'
import {Button} from 'native-base'
import {colors} from '../../utils/colors';
export default class LaunchScreen extends  React.Component{

    constructor(props){
        super(props)
        this.state={
            period:''


        }
    }

    render() {
        return (
            <View style={{flex:1,justifyContent:'center',alignItems:'center'}}>

                <Text style={{color:'black',fontSize:20}}>NY Times</Text>

                <TextInput

                    style={{ height :40,paddingStart:15,width:'80%',borderColor: 'gray', marginTop:20,borderWidth: 1, borderRadius:25}}
                    onChangeText={(period) => this.setState({period})}
                    value={this.state.period}
                    keyboardType={'numeric'}
                    placeholder={'Period'}
                />

                <Button primary style={{marginTop:40, width:'60%', borderRadius: 20, justifyContent:'center', alignItems:'center', backgroundColor:colors.primary}}
                  onPress={()=>{
                      let period=this.state.period
                      if (!period){
                          alert("Please enter period")
                      }else if (period==1 || period==7 || period==30) {
                          this.props.navigation.navigate('homeScreen',{period:period})


                      }else {
                          alert("Please enter either 1, 7 or 30.")
                      }
                  }}
                >
                    <Text style={{color:'white'}}>SUBMIT</Text>
                </Button>

            </View>
        );
    }

}
