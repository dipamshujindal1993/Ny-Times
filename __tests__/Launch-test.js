import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LaunchScreen from "../src/screens/launch/LaunchScreen";

test('Launch snapshot' , () => {
    const snap = renderer.create(
        <LaunchScreen/>
    ).toJSON();
    expect(snap).toMatchSnapshot();

});
