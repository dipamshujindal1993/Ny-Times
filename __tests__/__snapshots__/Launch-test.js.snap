// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 20,
      }
    }
  >
    NY Times
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Period"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 25,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 20,
        "paddingStart": 15,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#78E0C3",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 20,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginTop": 40,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "width": "60%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      SUBMIT
    </Text>
  </View>
</View>
`;
